### PR TITLE
fix compressedness flag check in verifymessage command

### PR DESCRIPTION
**Pull request by apoelstra**
fixed to match https://github.com/bitcoin/bitcoin/blob/master/src/pubkey.cpp#L191

### CODE REVIEW

File: src/bin/hal/cmd/message.rs [hunk 1]
```diff
@@ -116,7 +116,7 @@ fn exec_verify<'a>(matches: &clap::ArgMatches<'a>) {
 	}
 	let recid = secp256k1::recovery::RecoveryId::from_i32((sig_bytes[0] - 27 & 0x03) as i32)
 		.expect("invalid recoverable signature (invalid recid)");
-	let compressed = sig_bytes[0] & 0x04 != 0x04;
+	let compressed = (sig_bytes[0] - 27 & 0x04) != 0;
```
Comment (gwillen) [inline src/bin/hal/cmd/message.rs]: At least to my eye, parentheses would make this clearer. I suppose it matches what's above, though.

**Comment by elichai** [comment]
Yep. Looks good. I do agree about the parentheses though
(The same is here where I also tested it against core: https://github.com/elichai/bitsign/blob/master/src/main.rs#L171)

**Comment by stevenroose** [comment]
Eh, thanks!

**Comment by apoelstra** [comment]
I'm pretty sure if I put the parens in, rustc will complain about them being unnecessary.

**Pull request merged**